src/components/pages/units-page/edit-unit-dialog.tsx

```diff
@@ -246,7 +276,7 @@ export const EditUnitDialog = (props: EditUnitDialogProps) => {
     const body = {
       id,
       name,
-      musterConfiguration: !overrideDefault ? null : musterConfiguration.map(muster => {
+      musterConfiguration: musterConfigMode === MusterConfigMode.None ? [] : musterConfigMode === MusterConfigMode.Default ? null : musterConfiguration.map(muster => {
```

Comment:
This isn't quite as compact, but I think it's easier to read
```suggestion
      musterConfiguration: musterConfigMode === MusterConfigMode.None
        ? []
        : musterConfigMode === MusterConfigMode.Default
          ? null
          : musterConfiguration.map(muster => ({
            days: muster.days,
            startTime: muster.startTime,
            timezone: muster.timezone,
            durationMinutes: muster.durationMinutes,
          })),
```